browser/brave_vpn/dns/brave_vpn_dns_observer_service_unittest.cc

```diff
@@ -0,0 +1,317 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_vpn/dns/brave_vpn_dns_observer_service.h"
+
+#include <unordered_map>
+
+#include "base/run_loop.h"
+#include "base/test/bind.h"
+#include "brave/browser/brave_profile_prefs.h"
+#include "brave/components/brave_vpn/brave_vpn_utils.h"
+#include "brave/components/brave_vpn/pref_names.h"
+#include "chrome/browser/net/secure_dns_config.h"
+#include "chrome/browser/net/secure_dns_util.h"
+#include "chrome/browser/net/stub_resolver_config_reader.h"
+#include "chrome/browser/net/system_network_context_manager.h"
+#include "chrome/browser/prefs/browser_prefs.h"
+#include "chrome/common/pref_names.h"
+#include "chrome/test/base/scoped_testing_local_state.h"
+#include "chrome/test/base/testing_browser_process.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "content/public/test/browser_task_environment.h"
+#include "net/dns/public/secure_dns_mode.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_vpn {
+namespace {
+const char kCustomServersURLs[] =
+    "https://server1.com\nhttps://server2.com/{?dns}";
+const char kCloudflareDnsProviderURL[] =
+    "https://chrome.cloudflare-dns.com/dns-query";
+}  // namespace
+
+class BraveVpnDnsObserverServiceUnitTest : public testing::Test {
+ public:
+  BraveVpnDnsObserverServiceUnitTest() {}
+
+  void SetUp() override {
+    RegisterLocalState(local_state_.registry());
+    RegisterUserProfilePrefs(profile_pref_service_.registry());
+    brave_vpn::RegisterProfilePrefs(profile_pref_service_.registry());
+
+    profile_pref_service_.registry()->RegisterBooleanPref(
```

Comment:
I don't understand, shouldn't this already be called by `RegisterLocalState`?

Reply:
this is profile pref and registered inside [`BraveVpnDnsObserverFactory` ](https://github.com/brave/brave-core/pull/15989/files#diff-4787f8a39728a6d931feab7913a73efcb7925f61ea4db2028fc07e89e63d298fR57)it is easier to register it here separately for tests